### PR TITLE
rebuild against boost

### PR DESCRIPTION
**Pull request by cbouss**
Changes:
- increase build number
- add uriparser as dep
- remove libiconv as dep
- remove boost from run deps

https://github.com/libkml/libkml/tree/1.3.0
https://anaconda.atlassian.net/browse/PKG-2014

TODO: remove abs.yaml before merge

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -28,38 +25,22 @@ requirements:
...
-    # zlib removed here because package has run_exports for itself
-    #    https://github.com/AnacondaRecipes/zlib-feedstock/pull/2
-    # this req isn't actually necessary, as run_exports will add it.  Left here for reference.
-    # - zlib 1.2.8
-    # boost removed here because package has run_exports for itself
-    #     https://github.com/AnacondaRecipes/boost-cpp-feedstock/pull/1
-    # this req isn't actually necessary, as run_exports will add it.  Left here for reference.
-    # - boost 1.64.*
-    # version pinning is provided by a "pin_run_as_build" key in the user-wide conda_build_config.yaml
-    - libboost
```
Comment (M-Waszkiewicz-Anaconda) [inline recipe/meta.yaml:52]: so it uses boost header only libraries ? Is it safe assumption ?

**Review by JeanChristopheMorinPerso — APPROVED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (cbouss) [reply recipe/meta.yaml:52]: based on the cmakelist, this seems to be the case.

**Review by M-Waszkiewicz-Anaconda — APPROVED** [review]
LGTM

File: recipe/meta.yaml [hunk 2]
```diff
@@ -8,18 +8,15 @@ source:
   fn: libkml-{{ version }}.tar.gz
   url: https://github.com/libkml/libkml/archive/{{ version }}.tar.gz
   sha256: 8892439e5570091965aaffe30b08631fdf7ca7f81f6495b4648f0950d7ea7963
-  patches:
+  patches:              # [win]
     # Patches for vc14. If this works we should wrapit in a ifdef and send the patch upstream.
-    - empty_char.patch  # [win and py>35]
-    - uriparse.patch
```
Comment (skupriienko) [inline recipe/meta.yaml:14]: The file https://github.com/AnacondaRecipes/libkml-feedstock/blob/851fa541bc5f1565ac18ae1f70ab26a8997e8c72/recipe/uriparse.patch can be removed from the recipe folder too

**Review by skupriienko — APPROVED** [review]
LGTM but the outdated patch

**Pull request merged**